crates/sncast/src/lib.rs

```diff
@@ -217,22 +246,15 @@ async fn build_account(
         parse_number(&account_data.private_key)
             .context("Failed to convert private key to FieldElement")?,
     ));
-    let address = parse_number(&account_data.address).with_context(|| {
-        format!(
-            "Failed to convert account address = {} to FieldElement",
-            &account_data.address
-        )
-    })?;
+
+    let address = account_data.get_address_as_felt()?;
     verify_account_address(address, chain_id, provider).await?;
 
     let class_hash = account_data
         .class_hash
```

Comment:
In the current implementation, we cannot because `get_class_hash_as_felt` requires the class hash to be present, which isn't always necessary for account building. In fact, I'm not sure, but perhaps the only reason we retain it is because `get_account_encoding` has a `class_hash` and we must provide some value there.

Reply:
but it seems to me it does exactly the same thing as this snippet? I mean, if class hash is not present an error is thrown in both cases, right?

Reply:
But this would actually be legit answer to my first question here https://github.com/foundry-rs/starknet-foundry/pull/2101/files#r1596532371 if we used this in this place. Correct me if I'm wrong, but we already accept the argument as an Option, so technically we could just check here what is being returned by it (is it Ok or an Error), and if it's the latter we can do nothing (or print out some warning, if it makes sense). Or am I missing something abvious here?

Reply:
You are right, it can be done this way